src/os_handlers.eliom

```diff
@@ -207,9 +207,13 @@ let activation_handler_common ~akey =
      TODO: do not disconnect users if we relog them with the same userid.
   *)
   try%lwt
-    let%lwt {userid; email; validity; action = _; data = _; autoconnect} =
+    let%lwt {userid; email; validity; action; data = _; autoconnect} =
       Os_user.get_activationkey_info akey in
     let%lwt () =
+      if action = "activation" && validity <= 0L then
```

Comment:
I guess you can use a `` `Custom of string`` constructor for the user-provided actions. `` `Custom s`` can be stored as `s` in the DB, and when reading something from the DB, we will be using `` `Custom`` as a fallback case for the unknown actions.